pkg/util/filepath.go

```diff
@@ -69,8 +70,18 @@ func walk(path string, info os.FileInfo, resolvedPath string, symlinkPathsFollow
 			symlinkPathsFollowed[path2] = true
 		}
 		info2, err := os.Lstat(path2)
-		if err != nil {
-			return err
+		if err == nil {
+			return walk(path, info2, path2, symlinkPathsFollowed, walkFn)
+		}
+		var pathErr *os.PathError
+		if errors.As(err, &pathErr) && errors.Is(err, syscall.ENOENT) {
```

Comment:
Also toyed around with 

```go
if errors.As(err, &pathErr) && errors.Is(err, syscall.ENOENT) {
	p, err2 := filepath.EvalSymlinks(resolvedPath)
	if err2 == nil || !filepath.IsAbs(p) {
		return err
	}
	info2, err2 = os.Lstat(p)
	if err2 != nil {
		return err
	}
}
```
but that immediately resolves the symlink - and in cases where we point from 1 symlink to another, this will potentially skip over a file walk iteration, which is why I believe the current code is preferable